peer/peerdiscovery_test.go

```diff
@@ -0,0 +1,66 @@
+package peer_test
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/renproject/aw/wire"
+	"github.com/renproject/id"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("Peer", func() {
+	Context("when trying to discover other peers using the peer discovery client", func() {
+		It("should successfully find all peers", func() {
+			n := 5
+			opts, peers, tables, _, _, transports := setup(n)
+
+			for i := range peers {
+				peers[i].Receive(context.Background(), func(from id.Signatory, msg wire.Msg) error { return nil })
+			}
+			for i := range peers {
+				ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+				defer cancel()
+				go peers[i].Run(ctx)
+				tables[i].AddPeer(opts[(i+1)%n].PrivKey.Signatory(),
+					wire.NewUnsignedAddress(wire.TCP,
+						fmt.Sprintf("%v:%v", "localhost", uint16(3333+((i+1)%n))), uint64(time.Now().UnixNano())))
+			}
+
+			<-time.After(1 * time.Second)
+			for i := range peers {
+				ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
+				defer cancel()
+				go peers[i].DiscoverPeers(ctx)
+			}
+			<-time.After(4 * time.Second)
```

Comment:
```suggestion
			ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
			defer cancel()
			for i := range peers {
				go peers[i].DiscoverPeers(ctx)
			}
			<-ctx.Done()
```
It is better to rely on concrete moments in the test (e.g. "by the time the context has expired") than to rely on ad-hoc timeouts. This way, if someone extends the context duration, this test will still work as expected.